commit faa8427efc55a606f38296e18bf5fb6f9043d67e
Author: Rob Johnston <rob.johnston@joyent.com>
Date:   2019-06-05T11:24:54-07:00 (4 months ago)
    
    TRITON-1651 sdc-usbkey: set-variable should enclose value in quotes

diff --git a/tools/lib/usbkey.js b/tools/lib/usbkey.js
index 6c46adef..04a89aa9 100644
--- a/tools/lib/usbkey.js
+++ b/tools/lib/usbkey.js
@@ -1193,6 +1193,13 @@ set_variable_loader(mountpoint, name, value, callback)
     mod_assert.func(callback, 'callback');
 
     var search = '^\\s*' + name + '\\s*=\\s*.*$';
+
+    /*
+     * Make sure the value is enclosed in quotes (") in case the value contains
+     * embedded whitespace.  Otherwise, loader will fail to parse it correctly.
+     */
+    value = '"' + value + '"';
+
     var replace = name + '=' + value;
 
     sedfile(mountpoint + '/boot/loader.conf', search, replace, callback);
